osf/utils/migrations.py

```diff
@@ -413,6 +435,96 @@ def unmap_schemablocks(*args):
     RegistrationSchemaBlock.objects.all().delete()
 
 
+# For registration_responses validation
+def get_multiple_choice_options(registration_schema, question):
+    """
+    Returns a dictionary with an 'enum' key, and a value as
+    an array with the possible multiple choice answers for a given question.
+    Schema blocks are linked by schema_block_group_keys, so fetches multiple choice options
+    with the same schema_block_group_key as the given question
+    :question SchemaBlock with an registration_response_key
+    """
+    options = registration_schema.schema_blocks.filter(
+        schema_block_group_key=question.schema_block_group_key,
+        block_type='select-input-option'
+    ).values_list('display_text', flat=True)
+
+    options = list(options)
+    if '' not in options:
+        options.append('')
```

Comment:
@aaxelb, this line right here is going to cause problems.  If a field is not required, multiple choice questions need to accept an empty string for a time. But once it's required (when we're actually registering, and it's required), an empty string is not going to cut it.

Existing validation for `registration_metadata` only adds an empty string to the jsonschema if it's required, but we have less flexibility to be dynamic about this, because we're caching these schemas.

This makes me lean against caching.

Reply:
Hm yeah, that makes me lean against caching too. It's easy enough to build it on the fly, so we should probably revert to that. Might be generally good to consider caching a separate optimization task from whatever we're building.

Reply:
ok I'll see if I can pull this out really quick, apologies, I just noticed this yesterday.